provider/etcd/etcd_client.go

```diff
@@ -228,10 +275,13 @@ func (ec *EtcdClient) RunWatch(ctx context.Context) {
 
 		level.Info(ec.logger).Log("msg", "Etcd Watch Started")
 		for wresp := range rch {
+			etcdQueueLength.With(prometheus.Labels{"name": "watch"}).Set(float64(len(rch)))
+
 			for _, ev := range wresp.Events {
 				level.Debug(ec.logger).Log("msg", "watch received",
 					"type", ev.Type, "key", fmt.Sprintf("%q", ev.Kv.Key), "value", fmt.Sprintf("%q", ev.Kv.Value))
 				if ev.Type.String() == "PUT" {
+					etcdWatchOperationsTotal.With(prometheus.Labels{"operation": "put"}).Inc()
 					alert, err := UnmarshalAlert(string(ev.Kv.Value))
 					if err != nil {
 						continue
```

Comment:
MAY output some warning so that people know there is a problem when they check logs, maybe just the raw key/value from etcd.